mbus-api/src/v0.rs

```diff
@@ -475,3 +547,274 @@ impl From<i32> for ReplicaState {
         }
     }
 }
+
+/// Volume Nexuses
+///
+/// Get all the nexuses with a filter selection
+#[derive(Serialize, Deserialize, Default, Debug, Clone)]
+pub struct GetNexuses {
+    /// Filter request
+    pub filter: Filter,
+}
+
+/// Nexus information
+#[derive(Serialize, Deserialize, Default, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Nexus {
+    /// id of the mayastor instance
+    pub node: String,
+    /// name of the nexus
+    pub uuid: String,
+    /// size of the volume in bytes
+    pub size: u64,
+    /// current state of the nexus
+    pub state: NexusState,
+    /// array of children
+    pub children: Vec<Child>,
+    /// URI of the device for the volume (missing if not published).
+    /// Missing property and empty string are treated the same.
+    pub device_uri: String,
+    /// total number of rebuild tasks
+    pub rebuilds: u32,
+}
+
+/// Child information
+#[derive(Serialize, Deserialize, Default, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Child {
+    /// uri of the child device
+    pub uri: String,
+    /// state of the child
+    pub state: ChildState,
+    /// current rebuild progress (%)
+    pub rebuild_progress: Option<i32>,
+}
+
+/// Child State information
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
+pub enum ChildState {
```

Comment:
We need to think about this I guess. What is the single source of truth here for states? What is this state based on? the real state, the mapped state? And what if we changed the real state how will that brake code here? etc..

Reply:
Yeah... another option for these services is for us to use the actual state (not mapped one) from the get-go.

Reply:
I've raised CAS-608 for this.